src/components/date-picker/calendar.tsx

```diff
@@ -240,41 +288,90 @@ export class Calendar extends React.Component<CalendarProps, CalendarState> {
         }
     }
 
-    private toggleMonthView = () => {
-        this.setState({showMonthView: !this.state.showMonthView});
+    private toggleDateSelectView = () => {
+        if (this.state.showYearView && this.state.showMonthView) {
+            this.state.showYearView = false;
```

Comment:
This boolean flag game could be a bit much. consider if using an enum like ```this.state.view = ClendarViews.Year``` would make this shorter and more readable